public/pages/CreateIndex/components/IndexMapping/IndexMapping.tsx

```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { forwardRef, useCallback, useState, Ref, useRef, useMemo, useImperativeHandle } from "react";
+import { EuiTreeView, EuiIcon, EuiTreeViewProps, EuiButton, EuiSpacer, EuiButtonGroup, EuiLink } from "@elastic/eui";
+import { set, get, isEmpty } from "lodash";
+import JSONEditor, { IJSONEditorRef } from "../../../../components/JSONEditor";
+import { Modal } from "../../../../components/Modal";
+import { MappingsProperties, MappingsPropertiesObject } from "../../../../../models/interfaces";
+import CustomFormRow from "../../../../components/CustomFormRow";
+import MappingLabel, { IMappingLabelRef } from "../MappingLabel";
+import "./IndexMapping.scss";
+
+export const transformObjectToArray = (obj: MappingsPropertiesObject): MappingsProperties => {
+  return Object.entries(obj).map(([fieldName, fieldSettings]) => {
+    const { properties, ...others } = fieldSettings;
+    const payload: MappingsProperties[number] = {
+      ...others,
+      fieldName,
+    };
+    if (properties) {
+      payload.properties = transformObjectToArray(properties);
+    }
+    return payload;
+  });
+};
+
+export const transformArrayToObject = (array: MappingsProperties): MappingsPropertiesObject => {
+  return array.reduce((total, current) => {
+    const { fieldName, properties, ...others } = current;
+    const payload: MappingsPropertiesObject[string] = {
+      ...others,
+    };
+    if (properties) {
+      payload.properties = transformArrayToObject(properties);
+    }
+    return {
+      ...total,
+      [current.fieldName]: payload,
+    };
+  }, {} as MappingsPropertiesObject);
+};
+
+const countNodesInTree = (array: MappingsProperties) => {
+  return array.reduce((total, current) => {
+    total = total + 1;
+    const { properties } = current;
+    if (properties) {
+      total = total + countNodesInTree(properties);
+    }
+    return total;
+  }, 0);
+};
+
+export type IndexMappingsAll = {
```

Comment:
Move the types/interfaces to a separate file

Reply:
Fixed on https://github.com/opensearch-project/index-management-dashboards-plugin/commit/92e7db4f89ff5c00d3d5013c7a8e1756f3c9162c